lib/internal/modules/cjs/loader.js

```diff
@@ -792,24 +806,31 @@ function wrapSafe(filename, content) {
       } : undefined,
     });
   }
-
-  const compiled = compileFunction(
-    content,
-    filename,
-    0,
-    0,
-    undefined,
-    false,
-    undefined,
-    [],
-    [
-      'exports',
-      'require',
-      'module',
-      '__filename',
-      '__dirname',
-    ]
-  );
+  let compiled;
+  try {
+    compiled = compileFunction(
+      content,
+      filename,
+      0,
+      0,
+      undefined,
+      false,
+      undefined,
+      [],
+      [
+        'exports',
+        'require',
+        'module',
+        '__filename',
+        '__dirname',
+      ]
+    );
+  } catch (err) {
+    if (err instanceof SyntaxError) {
```

Comment:
can we remove this `if` / is there a reason to only check if it is a `SyntaxError`? My concern is about the reliability of `instanceof` cross Realm, a check of `err.name` or just excluding the test since we are already on a slower error path seems a bit better to me.

Reply:
removed the condition.